dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java

```diff
@@ -118,70 +121,109 @@ protected synchronized void decorateWithTLS(TLSClientParameters tlsClientParamet
 
         int hash = tlsClientParameters.hashCode();
         if (hash != lastTlsHash) {
+	    LOG.fine("Hash does not match!"); // Liberty Change
             lastTlsHash = hash;
             socketFactory = null;
         }
 
         // always reload socketFactory from HttpsURLConnection.defaultSSLSocketFactory and
         // tlsClientParameters.sslSocketFactory to allow runtime configuration change
+
         if (tlsClientParameters.isUseHttpsURLConnectionDefaultSslSocketFactory()) {
+            LOG.fine("tlsClientParameters property: isUseHttpsURLConnectionDefaultSslSocketFactory is set to true!"); // Liberty Change
             socketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
-
         } else if (tlsClientParameters.getSSLSocketFactory() != null) {
             // see if an SSLSocketFactory was set. This allows easy interop
             // with not-yet-commons-ssl.jar, or even just people who like doing their
             // own JSSE.
+            LOG.fine("Get SSL socketFactory from tlsClientParameters"); // Liberty Change
             socketFactory = tlsClientParameters.getSSLSocketFactory();
-
         } else if (socketFactory == null) {
-
             SSLContext ctx = null;
+            LOG.fine("SSL socketFactory is NULL"); // Liberty Change
             if (tlsClientParameters.getSslContext() != null) {
                 // Use the SSLContext which was set
+                LOG.fine("Get SSLContext from tlsClientParameters"); // Liberty Change
                 ctx = tlsClientParameters.getSslContext();
             } else {
-                // Create socketfactory with tlsClientParameters's Trust Managers, Key Managers, etc
-                ctx = org.apache.cxf.transport.https.SSLUtils.getSSLContext(tlsClientParameters);
+		// Liberty Change Start
+	        if (tlsClientParameters.isJaxrsClient() || (tlsClientParameters.getTrustManagers() != null && 
+			tlsClientParameters.getKeyManagers() != null)) {
+                   // Get sslcontext with tlsClientParameters's Trust Managers, Key Managers, etc
+                   LOG.fine("Get SSLContext with tlsClientParameters's Trust Managers, Key Managers");
+                   ctx = org.apache.cxf.transport.https.SSLUtils.getSSLContext(tlsClientParameters);
+	        }
+		else {
+                   LOG.fine("No trustManagers set on tlsClientParameters, so use Liberty's DefaultSSLSocketFactory");
+                   socketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
+		}
+		// Liberty Change End
             }
-
-            String[] cipherSuites =
-                SSLUtils.getCiphersuitesToInclude(tlsClientParameters.getCipherSuites(),
+	    if (ctx != null) { // do we have SSL context?
```

Comment:
Great idea to add SSL context check. It might be good to add trace showing when context is null?